### PR TITLE
Fix envoyfilter matching virtual inbound passthrough

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/30191

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go [hunk 1]
```diff
@@ -983,103 +982,6 @@ func TestApplyListenerPatches(t *testing.T) {
 		DownstreamNodes: []string{"foo"},
 	}
 	faultFilterOutAny, _ := ptypes.MarshalAny(faultFilterOut)
-	sidecarInboundIn := []*listener.Listener{
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch_test.go:986]: this test is not needed, we only have virtual inbound now.

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go [hunk 2]
```diff
@@ -592,6 +592,12 @@ func filterChainMatch(fc *xdslistener.FilterChain, cp *model.EnvoyFilterConfigPa
 			return false
 		}
 	}
+	isVirtual := listener.Name == VirtualInboundListenerName || listener.Name == VirtualOutboundListenerName
+	// We only do this for virtual listeners, which will move the listener port into a FCM. For non-virtual listeners,
+	// we will handle this in the proper listener match.
+	if isVirtual && cMatch.GetPortNumber() > 0 && fc.GetFilterChainMatch().GetDestinationPort().GetValue() != cMatch.GetPortNumber() {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go:598]: why donot you compare the port when it is non virtual listener

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go:598]: IC, listener match is done in a separate function. L 473

**Comment by hzxuzhonghu** [comment]
Should add a test for virtual listener

**Comment by howardjohn** [comment]
there already is one, it will fail without my pr with the extra "catch all"
filter chain I added

**Review by hzxuzhonghu — APPROVED** [review]
Thanks for the clarification.

**Pull request merged**